src/blocks/block-coauthors/edit.js

```diff
@@ -0,0 +1,243 @@
+		{ template : [['cap/coauthor-name']]}
+	) } />;
+}
+
+const ALLOWED_FORMATS = [
+	'core/bold',
+	'core/italic',
+	'core/text-color',
+];
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ *
+ * @return {WPElement} Element to render.
+ */
+export default function Edit( { attributes, setAttributes, clientId, context, isSelected } ) {
+
+	const { prefix, separator, lastSeparator, suffix, layout, textAlign } = attributes;
+	const { postId } = context;
+	const authorPlaceholder = useSelect( select => select( 'cap/blocks' ).getAuthorPlaceholder(), []);
+	const [ coAuthors, setCoAuthors ] = useState([authorPlaceholder]);
+	const [ activeBlockContextId, setActiveBlockContextId ] = useState();
+	const noticesDispatch = useDispatch('core/notices');
+
+	useEffect(()=>{
+		if ( ! postId ) {
+			return;
+		}
+
+		const controller = new AbortController();
+
+		apiFetch( {
+			path: `/coauthors-blocks/v1/coauthors/${postId}/`,
+			signal: controller.signal
+		} )
+		.then( setCoAuthors )
+		.catch( handleError )
+
+		return () => {
+			controller.abort();
+		}
+	},[postId]);
+
+	/**
+	 * Handle Error
+	 * 
+	 * @param {Error}
+	 */
+	function handleError( error ) {
+		if ( 'AbortError' === error.name ) {
+			return;
+		}
+		noticesDispatch.createErrorNotice( error.message, { isDismissible: true } );
+	}
+
+	const blocks = useSelect( (select) => {
+		return select( blockEditorStore ).getBlocks( clientId );
+	});
+
+	const setLayout = ( nextLayout ) => {
+		setAttributes( {
+			layout: { ...layout, ...nextLayout },
+		} );
+	}
+
+	const layoutControls = [
+		{
+			icon: list,
+			title: __( 'Inline' ),
+			onClick: () => setLayout( { type: 'inline' } ),
+			isActive: layout.type === 'inline',
+		},
+		{
+			icon: grid,
+			title: __( 'Block' ),
+			onClick: () =>
+				setLayout( { type: 'block' } ),
+			isActive: layout.type === 'block',
+		},
+	];
+
+	return (
+		<>
+			<BlockControls>
+				<ToolbarGroup controls={ layoutControls } />
+				<AlignmentControl
+					value={ textAlign }
+					onChange={ ( nextAlign ) => {
+						setAttributes( { textAlign: nextAlign } );
+					} }
+				/>
+			</BlockControls>
+			<div { ...useBlockProps({
+					className: classnames( {
+						[`is-layout-cap-${layout.type}`]: layout.type,
+						[`has-text-align-${ textAlign }`]: textAlign,
+					}),
+					style: {
+						gap: __experimentalGetGapCSSValue( attributes?.style?.spacing?.blockGap )
+					}
+				})
+			}>
+				{
+					coAuthors &&
+					'inline' === layout.type &&
+					( isSelected || prefix ) &&
+					(
+						<RichText
+							allowedFormats={ ALLOWED_FORMATS }
+							className="wp-block-cap-coauthors__prefix"
+							multiline={ false }
+							aria-label={ __( 'Prefix' ) }
+							placeholder={ __( 'Prefix' ) + ' ' }
```

Comment:
Where is this Prefix set as "By" by default?

Reply:
In the block.json